### PR TITLE
Simplify setting workspace metadata & add tests

### PR DESCRIPTION
**Pull request by MrCreosote**
... in preparation for setting object admin metadata which will hopefully reuse the code

### CODE REVIEW

File: src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java [hunk 1]
```diff
@@ -699,86 +699,67 @@ private void setCreatedWorkspacePermissions(
...
-	public Instant setWorkspaceMeta(final ResolvedWorkspaceID rwsi,
+	public Instant setWorkspaceMeta(
+			final ResolvedWorkspaceID rwsi,
 			final WorkspaceUserMetadata newMeta)
-			throws WorkspaceCommunicationException,
-			CorruptWorkspaceDBException {
-
+			throws WorkspaceCommunicationException, CorruptWorkspaceDBException {
+		// TODO NOW CODE integrate keys to remove (below) into this method and do it all in
+		// 					one shot
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java:707]: mysterious space?

File: src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java [hunk 2]
```diff
@@ -699,86 +699,67 @@ private void setCreatedWorkspacePermissions(
...
+					query,
+					new Document("$set", metaUpdate.append(Fields.WS_MODDATE, Date.from(time))));
+			if (ur.getModifiedCount() == 1) { //ok, it worked
+				return time;
+			} else if (attempts >= maxattempts) {
+				throw new WorkspaceCommunicationException(
+						String.format("Failed to update metadata %s times", attempts));
+			} else {
+				return null;
+			}
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java:758]: if you were having an anti-indentation day, you could do:

```
			if (ur.getModifiedCount() == 1) { //ok, it worked
 				return time;
 			}
 			if (attempts >= maxattempts) {
 				throw new WorkspaceCommunicationException(
 						String.format("Failed to update metadata %s times", attempts));
 			}
                        return null;
```

**Review by ialarmedalien — APPROVED** [review]

File: src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java (hunk 1 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java:707]: I tabbed it in to make it clear it's a continuation of the TODO

File: src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java (hunk 2 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java:758]: I actually prefer the `else`s here, makes things a bit easier to read for me. Not a big deal though

**Pull request merged**